R/standalone-types-check.R

```diff
@@ -531,4 +538,43 @@ check_data_frame <- function(x,
   )
 }
 
+check_numbers_decimal <- function(x,
+                                  ...,
+                                  min = NULL,
+                                  max = NULL,
+                                  allow_infinite = TRUE,
+                                  allow_na = FALSE,
+                                  allow_null = FALSE,
+                                  arg = caller_arg(x),
+                                  call = caller_env()) {
+  if (missing(x)) {
+    exit_code <- IS_NUMBER_false
+  } else if (0 == (exit_code <- max(map_dbl(x, \(x) .standalone_types_check_dot_call(
+    ffi_standalone_check_number_1.0.7,
+    x,
+    allow_decimal = TRUE,
+    min,
+    max,
+    allow_infinite,
+    allow_na,
+    allow_null
+  ))))) {
+    return(invisible(NULL))
+  }
+
+  .stop_not_number(
+    x,
+    ...,
+    exit_code = exit_code,
+    allow_decimal = TRUE,
+    vector = TRUE,
```

Comment:
Potential upgrade: Should probably pass the indices of the failing elements, as well as the reasons, so they can be reported in the error message. These should be truncated if there are too many.

But in a first pass it's fine to just fail saying all elements in the vector should follow the required invariants.